### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by simplay**
Fixing command `!ll A_NICK` and various runtime exceptions present after calling `!bb`

### CODE REVIEW

**Comment by simplay** [comment]
@rathrio: Run this locally and it seems to fix all the previousely present issues.

**Comment by rathrio** [comment]
will check this out in a couple of hours. time for TS chat later?

**Comment by simplay** [comment]
:+1:

File: src/user.rb [hunk 1]
```diff
@@ -91,6 +104,15 @@ def bot?
     @levels.values.include?("Admin Server Query")
   end
 
+  # Check whether this user is neither a bot nor the nil_user.
+  #
+  # @info: User instances that yield true correspond to real human clients.
+  # @return [Boolean] true if user is a human client otherwise false.
+  def exists?
```
Comment (rathrio) [inline src/user.rb]: how about we call this `human?`.

`#bot?`, `#human?`, and `#sentinel?` would be really nice.

**Comment by simplay** [comment]
@rathrio: I applied the changes according to your feedback. 
I reverted the mutex stuff. When introducing it, I was thinking on making the bot internally multithreaded (processing commands, stored in a queue). I wanted to ensure that the bot's internal state was thread save. Since a bot is internally not concurrent, i decided to revert these changes. Sorry for confusing you and thanks for commenting.

**tl&dr**: Ready for a review. Tested these changes locally and everything seems to work.

**Comment by rathrio** [comment]
@simplay haha ok. It's fine. I'm easy to confuse.

Runs just fine :+1:

**Pull request merged**